src/ScreenWidget.py

```diff
@@ -1,160 +1,94 @@
-        # takes a 3000,4000 grayscale image straight from the camera
-
-        self.data = data
-        rgbData = np.zeros((HEIGHT_FRAME,WIDTH_FRAME,4), dtype=np.uint8)
-        for i in range(3):
-            rgbData[:,:,i] = self.data
-        self.frame = QImage(rgbData, rgbData.shape[1], rgbData.shape[0],
-                                QImage.Format_RGB32)
-
-        self.display()
-
-    def display(self):
-        if self.zoom:
-            self.display_zoom()
-        else:
-            self.display_scaled()
-
-    def display_scaled(self):
-        self.painter.drawImage(0,0, self.frame.scaled(WS, HS, Qt.IgnoreAspectRatio,
-                                Qt.SmoothTransformation))
-        self.painter.drawImage(0,0, self.overlay.scaled(WS, HS, Qt.IgnoreAspectRatio,
-                                Qt.SmoothTransformation))
-        self.setPixmap(self.pixmap)
-        self.update()
-
-    def display_zoom(self):
-        rect = QRect(int(self.xc-WS/2), int(self.yc-HS/2), WS, HS)
-        self.painter.drawImage(0,0, self.frame.copy(rect))
-        self.painter.drawImage(0,0, self.overlay.copy(rect))
-        self.setPixmap(self.pixmap)
-        self.update()
-
-    def updatePixel(self, x, y): 
-        self.overlay.fill(qRgba(0,0,0,0))
-        # ^ this could be faster if we just qfill the known previous pixel
-        for i in range(x-8, x+9):
-            for j in range(y-8, y+9):
-                if ((i>=(x-1)) and (i<=(x+1))) or ((j>=(y-1)) and (j<=(y+1))):
-                    # 3-pix red crosshair
-                    self.overlay.setPixel(i, j, qRgba(255, 0, 0, 255))
-        self.display()
+        self.imageItem.setImage(data)
+
+    def updateCameraMenu(self):
+        for act in self.cameraActions:
+            act.triggered.disconnect(act.callback)
+            self.viewBox.menu.removeAction(act)
+        for camera in self.model.cameras:
+            act = QAction(camera.name())
+            act.callback = functools.partial(self.setCamera, camera)
+            act.triggered.connect(act.callback)
+            self.cameraActions.append(act)
+            self.viewBox.menu.insertAction(self.cameraActionSeparator, act)
+
+    def imageClicked(self, event):
+        if event.button() == QtCore.Qt.MouseButton.LeftButton:            
+            self.clickTarget.setPos(event.pos())
+            self.clickTarget.setVisible(True)
 
     def setCamera(self, camera):
         self.camera = camera
         self.refresh()
 
-    def mousePressEvent(self, e): 
-
-        if e.button() == Qt.LeftButton:
-            self.xsel = e.x()
-            self.ysel = e.y()
-            if self.zoom:
-                self.xsel = int(e.x() + self.xc - WS/2)
-                self.ysel = int(e.y() + self.yc - HS/2)
-            else:
-                self.xsel = int(e.x() * CONVERSION_PX)
-                self.ysel = int(e.y() * CONVERSION_PX)
-            self.updatePixel(self.xsel, self.ysel)
-            self.selected.emit(self.xsel, self.ysel)
```

Comment:
the emission of this signal sets the L and R correspondence points in the Model, so removing this breaks calibration, triangulation etc.